drivers/SmartThings/zwave-switch/src/multichannel-device/init.lua

```diff
@@ -0,0 +1,93 @@
+-- Copyright 2022 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+local cc = require "st.zwave.CommandClass"
+local capabilities = require "st.capabilities"
+local st_device = require "st.device"
+local MultiChannel = (require "st.zwave.CommandClass.MultiChannel")({ version = 3 })
+
+local map_device_class_to_profile = {
+  [0x10] = "metering-switch",
+  [0x31] = "metering-switch",
+  [0x11] = "metering-dimmer",
+  [0x08] = "metering-sensor",
+  [0x21] = "metering-sensor",
+  [0x20] = "metering-sensor",
+  [0xA1] = "metering-sensor"
+}
+
+local function can_handle_multichannel_device(opts, driver, device, ...)
+  return device:supports_capability(capabilities.zwMultichannel)
+end
+
+local function find_child(device, src_channel)
+  if src_channel == 0 then
+    return device
+  else
+    return device:get_child_by_parent_assigned_key(string.format("%02X", src_channel))
+  end
+end
+
+local function device_init(driver, device)
+  if device.network_type == st_device.NETWORK_TYPE_ZWAVE then
+      device:set_find_child(find_child)
+  end
+end
+
+local function get_profile(generic_device_class)
+  local device_class = generic_device_class or 0x10
+  return map_device_class_to_profile[device_class] or "metering_switch"
+end
+
+local function prepare_metadata(device, endpoint, generic_device_class)
+  local name = string.format("%s %d", device.label, endpoint)
+  return {
+    type = "EDGE_CHILD",
+    label = name,
+    profile = get_profile(generic_device_class),
+    parent_device_id = device.id,
+    parent_assigned_child_key = string.format("%02X", endpoint),
+    vendor_provided_label = name
+  }
+end
+
+local function device_added(driver, device)
+  if device.network_type ~= st_device.NETWORK_TYPE_CHILD  then
+    for index, endpoint in pairs(device.zwave_endpoints) do
+      device:send(MultiChannel:CapabilityGet({ end_point = index }))
+    end
+  end
+  device:refresh()
+end
+
+local function capability_get_report_handler(driver, device, cmd)
+  if find_child(device, cmd.args.end_point) == nil then
+      driver:try_create_device(prepare_metadata(device, cmd.args.end_point, cmd.args.generic_device_class))
+  end
+end
+
+local multichannel_device = {
+  NAME = "Z-Wave Device Multichannel",
+  lifecycle_handlers = {
+    init = device_init,
+    added = device_added
+  },
+  zwave_handlers = {
+    [cc.MULTI_CHANNEL] = {
+      [MultiChannel.CAPABILITY_REPORT] = capability_get_report_handler
```

Comment:
I'd like to see a test of this handler.

Reply:
I've created 2 unit tests (one for the parent device which has the child, and one for parent without children) and for the second one the unit tests fails with following stacktrace: 
```
INFO  || <ZwaveDevice: 00000000-1111-2222-3333-000000000001 [01] (Z-Wave Switch Multichannel)> received Z-Wave command: {args={command_class="", dynamic=false, end_point=1, generic_device_class=16, specific_device_class=0}, cmd_class="MULTI_CHANNEL", cmd_id="CAPABILITY_REPORT", dst_channels={}, encap="AUTO", payload="\x01\x10\x00", src_channel=0, version=2}
TRACE || Found ZwaveDispatcher handler in zwave_switch -> Z-Wave Device Multichannel
Received unexpected error: ...ojects\SmartThingsEdgeDrivers\lua_libs-api_v1\cosock.lua:296: ...SmartThingsEdgeDrivers\lua_libs-api_v1\st\dispatcher.lua:233: Error encountered while processing event for <ZwaveDevice: 00000000-1111-2222-3333-000000000001 [01] (Z-Wave Switch Multichannel)>:
    arg1: {args={command_class="", dynamic=false, end_point=1, generic_device_class=16, specific_device_class=0}, cmd_class="MULTI_CHANNEL", cmd_id="CAPABILITY_REPORT", dst_channels={}, encap="AUTO", payload="\x01\x10\x00", src_channel=0, version=2}
...SmartThingsEdgeDrivers\lua_libs-api_v1\st\dispatcher.lua:233: Error encountered while processing event for <ZwaveDevice: 00000000-1111-2222-3333-000000000001 [01] (Z-Wave Switch Multichannel)>:
    arg1: {args={command_class="", dynamic=false, end_point=1, generic_device_class=16, specific_device_class=0}, cmd_class="MULTI_CHANNEL", cmd_id="CAPABILITY_REPORT", dst_channels={}, encap="AUTO", payload="\x01\x10\x00", src_channel=0, version=2}
...cts\SmartThingsEdgeDrivers\lua_libs-api_v1\st\driver.lua:424: attempt to call a nil value (field 'create_device')
stack traceback:
	[C]: in function 'error'
	...cts\SmartThingsEdgeDrivers\lua_libs-api_v1\st\thread.lua:76: in function <...cts\SmartThingsEdgeDrivers\lua_libs-api_v1\st\thread.lua:54>
stack traceback:
	[C]: in function 'error'
	...ojects\SmartThingsEdgeDrivers\lua_libs-api_v1\cosock.lua:296: in function 'cosock.run'
	...cts\SmartThingsEdgeDrivers\lua_libs-api_v1\st\driver.lua:758: in function 'st.driver.run'
	./init.lua:147: in main chunk
	[C]: in function 'require'
	[C]: in function 'xpcall'
	...ngsEdgeDrivers\lua_libs-api_v1\integration_test\init.lua:220: in upvalue 'run_configured_test'
	...ngsEdgeDrivers\lua_libs-api_v1\integration_test\init.lua:336: in function 'integration_test.run_registered_tests'
	...hings\zwave-switch\src\test\test_multichannel_device.lua:2309: in main chunk
	[C]: in ?
FAILED
```

It is unable to call `create_device'. You can see it in next commit I've pushed.

Reply:
In this case, perhaps you should comment out the test and leave it with a TODO to address after the 046x release of the lua libraries (hopefully such a thing can get in that release, but no guarantees). We will have to test it manually if possible.

Reply:
That sounds good. Thanks @cjswedes